core/tauri-build/src/mobile.rs

```diff
@@ -48,13 +53,27 @@ dependencies {"
 
   app_build_gradle.push_str("\n}");
 
+  if let Some(version) = config.version.as_ref() {
+    app_tauri_properties.push_str(&format!("tauri.android.versionName={}\n", version));
+    if let Some(version_code) = config.bundle.android.version_code.as_ref() {
+      app_tauri_properties.push_str(&format!("tauri.android.versionCode={}\n", version_code));
+    } else if let Ok(version) = Version::parse(version) {
+      let version_code = version.major * 1000000 + version.minor * 1000 + version.patch;
+      app_tauri_properties.push_str(&format!("tauri.android.versionCode={}\n", version_code));
+    }
+  }
+
   write(&gradle_settings_path, gradle_settings).context("failed to write tauri.settings.gradle")?;
 
   write(&app_build_gradle_path, app_build_gradle)
     .context("failed to write tauri.build.gradle.kts")?;
 
+  write(&app_tauri_properties_path, app_tauri_properties)
```

Comment:
maybe avoid writing if `config.version` is None and don't emit `cargo:rerun` either?